tests/kernel/numa_irqbalance.pm

```diff
@@ -0,0 +1,113 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This modules verifies we have correct spreading of NVMe
+# interrupts across all NUMA nodes, especially when the number of
+# nodes exceeds the number of queues the device has.
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $self   = shift;
+    my $numcpu = get_required_var('QEMUCPUS');
+    my @interrupts;
+    my @irqs;         # number of interrupts by cpu
+    my $numqueues;    # number of NVMe queues, each queue has it's own MSI
+    my $fail = 0;
+
+    $self->select_serial_terminal;
+
+    $numqueues = script_output('grep -c nvme /proc/interrupts');
+    diag("The VM has $numqueues NVMe queues and $numcpu CPUs");
```

Comment:
would it be also useful to insert an INFO message? the diag ends up in "autoinst-log.txt", which is ok, but plenty of many other useless messages for this test.

Reply:
I don't know. I can add it, but this is just the variables you configure. I prefer just having the diag in the logs to search for when the test fails. I can also add an info, if you think it is useful.